server/base/src/main/java/org/apache/accumulo/server/tables/IllegalTableTransitionException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.tables;
+
+import org.apache.accumulo.core.master.state.tables.TableState;
+
+public class IllegalTableTransitionException extends RuntimeException {
+  private static final long serialVersionUID = 1L;
+
+  public IllegalTableTransitionException(TableState oldState, TableState newState) {
+    super("Error transitioning from " + oldState + " state to " + newState + " state");
+  }
+
+}
```

Comment:
I like the breakout into its own class.  We seem to have a lot of inner static classes for no good reason.  I think they only create clutter within the outer class and undermine some of the abstraction you get by creating the object in the first place.

Reply:
I agree. However, I was thinking about moving it back in to reduce the PR size a bit. I originally moved it out, because I was planning on moving it to `core/clientImpl` so I could have it thrown from one of the `retryLoop` methods in `ZooReader`. However, after moving it out, I realized all I really needed to do was make it a `RuntimeException`. I'm inclined to leave it now, just because I already simplified it a lot, and it is nice not to have a bunch of clutter.